<a href="https://colab.research.google.com/github/kelleyl54/DSPS_LKelley/blob/master/HW3_lKelley/Sept11exercise_lKelley.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [0]:
pd.read_csv?

In [0]:
#reading in the file from another source
rain=pd.read_csv("https://raw.githubusercontent.com/fedhere/DSPS/master/labs/1865331.csv")

In [0]:
rain.describe()

PRCP        SNOW        SNWD        WESD        WESF
count  3538.000000  2266.00000  216.000000  175.000000  256.000000
mean      0.137340     0.09594    1.026852    0.160571    0.073828
std       0.382174     0.64966    2.738447    0.457994    0.175507
min       0.000000     0.00000    0.000000    0.000000    0.000000
25%       0.000000     0.00000    0.000000    0.000000    0.000000
50%       0.000000     0.00000    0.000000    0.000000    0.000000
75%       0.070000     0.00000    0.350000    0.100000    0.100000
max       6.960000    12.30000   16.400000    2.400000    1.400000

In [0]:
rain.head(5)

STATION                    NAME        DATE  ...  SNWD  WESD  WESF
0  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-01  ...   0.0   NaN   NaN
1  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-02  ...   NaN   NaN   NaN
2  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-03  ...   NaN   NaN   NaN
3  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-04  ...   NaN   NaN   NaN
4  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-05  ...   NaN   NaN   NaN

[5 rows x 8 columns]

Notes: describe only works on columns with numerical types




Extracting some of the columns:

In [0]:
rain["DATE"].values

array(['2010-01-01', '2010-01-02', '2010-01-03', ..., '2019-09-06',
       '2019-09-07', '2019-09-08'], dtype=object)

In [0]:
#culling the data to only useful columns
rainonly = rain[["DATE","PRCP"]]

In [0]:
rainonly.DATE.dtype

dtype('O')

In [0]:
#turning the date data into more manageable datetime data
rainonly['date']=pd.to_datetime(rainonly.DATE)

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [0]:
rainonly.head()

DATE  PRCP       date
0  2010-01-01  0.28 2010-01-01
1  2010-01-02  0.00 2010-01-02
2  2010-01-03  0.00 2010-01-03
3  2010-01-04  0.00 2010-01-04
4  2010-01-05  0.00 2010-01-05

In [0]:
#broadcasting the data for after a certain datetime object
recent=rainonly[rainonly['date']>pd.to_datetime("2018/01/01")]

In [0]:
recent.describe()

PRCP
count  615.000000
mean     0.162504
std      0.365007
min      0.000000
25%      0.000000
50%      0.000000
75%      0.120000
max      2.450000

In [0]:
#taking the values from the new data and defining the parameters I will run the z-test with with a confidence interval of 2-sigma
samplemean=.162504
samplecount=615.0
popmean=.137
popstd=.38
s=popstd/(np.sqrt(615))

In [0]:
#creating the z value
Z=(samplemean-popmean)/s

In [0]:
#comparing the z value to the confidence interval
Z<2

True

In [0]:
#returning the actual z value
Z

1.664417452430633

The Z value returns 1.66, in unit of standard deviations, so if we choose a confidence level of 2 sigma, 1.66 < 2. The null hypothesis cannot be rejected for this data set, for whatever that should be in this simple example.